05_stringer/jessodri/main_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("127.0.0.1\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("actual: %s does not match expected: %s", actual, expected)
+	}
+}
+func TestStringMethod(t *testing.T) {
+	testCases := []struct {
+		description string
+		ip          IPAddr
+		expected    string
+	}{
+		{"happy path", IPAddr{127, 0, 0, 1}, "127.0.0.1"},
+		{"one value", IPAddr{127}, "127.0.0.0"},
+		{"two value", IPAddr{127, 0}, "127.0.0.0"},
+		{"three value", IPAddr{127, 0, 0}, "127.0.0.0"},
+		{"empty", IPAddr{}, "0.0.0.0"},
+	}
+	for _, tc := range testCases {
```

Comment:
Alternatively if you use the `map[string]struct{input IPAddr, expected string}` you can use the key to describe the test name and you don't have to access it from the struct.

```go
for name, tc := range testCases {
    t.Run(name, func(t *testing.T) { ... })
}
```